uber/models/__init__.py

```diff
@@ -376,47 +376,54 @@ def apply(self, params, *, bools=(), checkgroups=(), restricted=True, ignore_csr
         checkgroups = self.regform_checkgroups if restricted else checkgroups
         for column in self.__table__.columns:
             if (not restricted or column.name in self.unrestricted) and column.name in params and column.name != 'id':
-                if column.type is JSON or isinstance(column.type, JSON):
-                    value = params[column.name]
-                elif isinstance(params[column.name], list):
+                if isinstance(params[column.name], list):
                     value = ','.join(map(str, params[column.name]))
-                elif isinstance(params[column.name], bool):
-                    value = params[column.name]
-                elif params[column.name] is None:
-                    value = None
                 else:
-                    value = str(params[column.name]).strip()
+                    value = params[column.name]
```

Comment:
All this stuff under `apply()` is unrelated. This fixes a totally unrelated bug I discovered during testing.

Reply:
Ah, interesting.  What was the bug?

No real objection to removing the whitespace stripping - people would occasionally have leading or trailing spaces on fields, and I added the strip to clean it up.  This code does look cleaner overall, especially since it looks like there were 3 different if-checks which all just did the same thing.

Reply:
Not so much of a bug, really. More like an inefficiency. I was doing some load testing by generating thousands of promo codes at a time. I was seeing thousands of lines like this:
```
2018-08-11 21:10:04,353 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
2018-08-11 21:10:04,354 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
2018-08-11 21:10:04,355 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
2018-08-11 21:10:04,356 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
2018-08-11 21:10:04,358 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
2018-08-11 21:10:04,359 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
2018-08-11 21:10:04,361 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
2018-08-11 21:10:04,363 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
2018-08-11 21:10:04,365 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
2018-08-11 21:10:04,366 [DEBUG] uber.models.promo_code.PromoCode: Ignoring error coercing value for column promo_code.expiration_date: Not naive datetime (tzinfo is already set)
```

After some investigation, I realized were manipulating the params before calling `apply()`. We were setting `expiration_date = c.ESCHATON` if it wasn't set. Then the `apply()` function was converting the date to a string, and re-parsing that string. Of course, the string contained a timezone, so when we tried to localize the date we received that warning.

This change makes it so we don't automatically convert every param to a string before attempting to coerce it to the proper type.

I like the idea of stripping string params to clean them, so I'll put that back in!